File: swif/uno/modules/lengthOfLongestSubstring.swift
import Foundation

class LongestSubstringWithoutRepeatingCharacters {
    func lengthOfLongestSubstring(_ s: String) -> Int {
        var longest = 0
        var start = 0
        var end = 0
        var chars = [Character: Int]()
        for (i, c) in s.enumerated() {
            if let index = chars[c] {
                start = max(start, index + 1)
            }
            chars[c] = i
            end = i
            longest = max(longest, end - start + 1)
        }
        return longest
    }
}

File: Go/uno.com/pkg/LongestSubstringWithoutRepeatingCharacters/lengthOfLongestSubstring.go
package LongestSubstringWithoutRepeatingCharacters

func lengthOfLongestSubstring(s string) int {
	var max int
	var start int
	var end int
	var length int
	var charMap = make(map[byte]int)
	for i := 0; i < len(s); i++ {
		charMap[s[i]] = -1
	}
	for end < len(s) {
		if charMap[s[end]] >= start {
			start = charMap[s[end]] + 1
		}
		charMap[s[end]] = end
		length = end - start + 1
		if length > max {
			max = length
		}
		end++
	}
	return max
}